### ISSUE TITLE
Error After Install PrimeNG

### ISSUE THREAD

**Issue by mfalzetta** [sudheerj/generator-jhipster-primeng#39]
Hi,

I seek this tuto: https://www.primefaces.org/jhipster-with-primeng/

After install my jHipster app show many erros below.

U can help me?

```
ERROR in ./src/main/webapp/app/primeng/data/datascroller/datascrollerdemo.module.ts
Module not found: Error: Can't resolve '@angular/http' in '/Users/mfalzetta/Dropbox/MFalzetta/Projetos/Matta-Tecnologia/Security/Desenvolvimento/security-web/src/main/webapp/app/primeng/data/datascroller'
 @ ./src/main/webapp/app/primeng/data/datascroller/datascrollerdemo.module.ts 12:13-37
 @ ./src/main/webapp/app/primeng/primeng.module.ts
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts
 @ multi (webpack)-dev-server/client?http://localhost:9060 webpack/hot/dev-server ./src/main/webapp/app/app.main

ERROR in ./src/main/webapp/app/primeng/data/datatable/datatabledemo.module.ts
Module not found: Error: Can't resolve '@angular/http' in '/Users/mfalzetta/Dropbox/MFalzetta/Projetos/Matta-Tecnologia/Security/Desenvolvimento/security-web/src/main/webapp/app/primeng/data/datatable'
 @ ./src/main/webapp/app/primeng/data/datatable/datatabledemo.module.ts 13:13-37
 @ ./src/main/webapp/app/primeng/primeng.module.ts
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts
 @ multi (webpack)-dev-server/client?http://localhost:9060 webpack/hot/dev-server ./src/main/webapp/app/app.main

ERROR in ./src/main/webapp/app/primeng/data/dataview/dataviewdemo.module.ts
Module not found: Error: Can't resolve '@angular/http' in '/Users/mfalzetta/Dropbox/MFalzetta/Projetos/Matta-Tecnologia/Security/Desenvolvimento/security-web/src/main/webapp/app/primeng/data/dataview'
 @ ./src/main/webapp/app/primeng/data/dataview/dataviewdemo.module.ts 12:13-37
 @ ./src/main/webapp/app/primeng/primeng.module.ts
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts
 @ multi (webpack)-dev-server/client?http://localhost:9060 webpack/hot/dev-server ./src/main/webapp/app/app.main

ERROR in ./src/main/webapp/app/primeng/data/orderlist/orderlistdemo.module.ts
Module not found: Error: Can't resolve '@angular/http' in '/Users/mfalzetta/Dropbox/MFalzetta/Projetos/Matta-Tecnologia/Security/Desenvolvimento/security-web/src/main/webapp/app/primeng/data/orderlist'
 @ ./src/main/webapp/app/primeng/data/orderlist/orderlistdemo.module.ts 13:13-37
 @ ./src/main/webapp/app/primeng/primeng.module.ts
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts
 @ multi (webpack)-dev-server/client?http://localhost:9060 webpack/hot/dev-server ./src/main/webapp/app/app.main

ERROR in ./src/main/webapp/app/primeng/data/orgchart/orgchartdemo.module.ts
Module not found: Error: Can't resolve '@angular/http' in '/Users/mfalzetta/Dropbox/MFalzetta/Projetos/Matta-Tecnologia/Security/Desenvolvimento/security-web/src/main/webapp/app/primeng/data/orgchart'
 @ ./src/main/webapp/app/primeng/data/orgchart/orgchartdemo.module.ts 14:13-37
 @ ./src/main/webapp/app/primeng/primeng.module.ts
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts
 @ multi (webpack)-dev-server/client?http://localhost:9060 webpack/hot/dev-server ./src/main/webapp/app/app.main

ERROR in ./src/main/webapp/app/primeng/data/picklist/picklistdemo.module.ts
Module not found: Error: Can't resolve '@angular/http' in '/Users/mfalzetta/Dropbox/MFalzetta/Projetos/Matta-Tecnologia/Security/Desenvolvimento/security-web/src/main/webapp/app/primeng/data/picklist'
 @ ./src/main/webapp/app/primeng/data/picklist/picklistdemo.module.ts 12:13-37
 @ ./src/main/webapp/app/primeng/primeng.module.ts
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts
 @ multi (webpack)-dev-server/client?http://localhost:9060 webpack/hot/dev-server ./src/main/webapp/app/app.main

ERROR in ./src/main/webapp/app/primeng/data/schedule/scheduledemo.module.ts
Module not found: Error: Can't resolve '@angular/http' in '/Users/mfalzetta/Dropbox/MFalzetta/Projetos/Matta-Tecnologia/Security/Desenvolvimento/security-web/src/main/webapp/app/primeng/data/schedule'
 @ ./src/main/webapp/app/primeng/data/schedule/scheduledemo.module.ts 13:13-37
 @ ./src/main/webapp/app/primeng/primeng.module.ts
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts
 @ multi (webpack)-dev-server/client?http://localhost:9060 webpack/hot/dev-server ./src/main/webapp/app/app.main

ERROR in ./src/main/webapp/app/primeng/data/table/tabledemo.module.ts
Module not found: Error: Can't resolve '@angular/http' in '/Users/mfalzetta/Dropbox/MFalzetta/Projetos/Matta-Tecnologia/Security/Desenvolvimento/security-web/src/main/webapp/app/primeng/data/table'
 @ ./src/main/webapp/app/primeng/data/table/tabledemo.module.ts 13:13-37
 @ ./src/main/webapp/app/primeng/primeng.module.ts
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts
 @ multi (webpack)-dev-server/client?http://localhost:9060 webpack/hot/dev-server ./src/main/webapp/app/app.main

ERROR in ./src/main/webapp/app/primeng/data/tree/treedemo.module.ts
Module not found: Error: Can't resolve '@angular/http' in '/Users/mfalzetta/Dropbox/MFalzetta/Projetos/Matta-Tecnologia/Security/Desenvolvimento/security-web/src/main/webapp/app/primeng/data/tree'
 @ ./src/main/webapp/app/primeng/data/tree/treedemo.module.ts 13:13-37
 @ ./src/main/webapp/app/primeng/primeng.module.ts
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts
 @ multi (webpack)-dev-server/client?http://localhost:9060 webpack/hot/dev-server ./src/main/webapp/app/app.main

ERROR in ./src/main/webapp/app/primeng/data/treetable/treetabledemo.module.ts
Module not found: Error: Can't resolve '@angular/http' in '/Users/mfalzetta/Dropbox/MFalzetta/Projetos/Matta-Tecnologia/Security/Desenvolvimento/security-web/src/main/webapp/app/primeng/data/treetable'
 @ ./src/main/webapp/app/primeng/data/treetable/treetabledemo.module.ts 13:13-37
 @ ./src/main/webapp/app/primeng/primeng.module.ts
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts
 @ multi (webpack)-dev-server/client?http://localhost:9060 webpack/hot/dev-server ./src/main/webapp/app/app.main

ERROR in ./src/main/webapp/app/primeng/misc/blockui/blockuidemo.module.ts
Module not found: Error: Can't resolve '@angular/http' in '/Users/mfalzetta/Dropbox/MFalzetta/Projetos/Matta-Tecnologia/Security/Desenvolvimento/security-web/src/main/webapp/app/primeng/misc/blockui'
 @ ./src/main/webapp/app/primeng/misc/blockui/blockuidemo.module.ts 14:13-37
 @ ./src/main/webapp/app/primeng/primeng.module.ts
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts
 @ multi (webpack)-dev-server/client?http://localhost:9060 webpack/hot/dev-server ./src/main/webapp/app/app.main

ERROR in [at-loader] ./src/main/webapp/app/primeng/data/datascroller/datascrollerdemo.module.ts:4:28 
    TS2307: Cannot find module '@angular/http'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/data/datatable/datatabledemo.module.ts:5:28 
    TS2307: Cannot find module '@angular/http'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/data/dataview/dataviewdemo.module.ts:4:28 
    TS2307: Cannot find module '@angular/http'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/data/orderlist/orderlistdemo.module.ts:5:28 
    TS2307: Cannot find module '@angular/http'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/data/orgchart/orgchartdemo.module.ts:6:26 
    TS2307: Cannot find module '@angular/http'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/data/picklist/picklistdemo.module.ts:5:28 
    TS2307: Cannot find module '@angular/http'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/data/schedule/scheduledemo.module.ts:5:28 
    TS2307: Cannot find module '@angular/http'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/data/table/tabledemo.module.ts:5:28 
    TS2307: Cannot find module '@angular/http'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/data/tree/treedemo.module.ts:5:28 
    TS2307: Cannot find module '@angular/http'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/data/treetable/treetabledemo.module.ts:5:28 
    TS2307: Cannot find module '@angular/http'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/fileupload/fileupload/backend/fake-backend.ts:19:110 
    TS2307: Cannot find module '@angular/http'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/fileupload/fileupload/backend/fake-backend.ts:20:43 
    TS2307: Cannot find module '@angular/http/testing'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/fileupload/fileupload/fileuploaddemo.module.ts:13:26 
    TS2307: Cannot find module '@angular/http'.

ERROR in [at-loader] ./src/main/webapp/app/primeng/misc/blockui/blockuidemo.module.ts:7:26 
    TS2307: Cannot find module '@angular/http'.
Child html-webpack-plugin for "index.html":
     1 asset
    [./node_modules/html-webpack-plugin/lib/loader.js!./src/main/webapp/index.html] ./node_modules/html-webpack-plugin/lib/loader.js!./src/main/webapp/index.html 1.53 kB {0} [built]
            factory:110ms building:1002ms = 1112ms
webpack: Failed to compile.
[Browsersync] Proxying: http://localhost:9060
[Browsersync] Access URLs:
 -------------------------------------
       Local: http://localhost:9000
```

**Comment by nomahnivek** [comment]
This is happening because @angular/http has been replaced by @angular/common/http.  
I'm fixing this for myself right now. So far I have not found any component that actually need the HttpModule module. I'll update this with more info when I'm done updating all of the files.

**Comment by normyx** [comment]
Hi,
I have the same issue with 4.14.4.
Even if I am manually @angular/http (npm install ...), I have other issues.

**Comment by sudheerj** [comment]
I can able to successfully run the latest Jhipster-PrimeNG application with JHipster 4.13.0 version. Can you please confirm that it specific to 4.14.x version. So that I try to replicate with latest version of JHipster

**Comment by normyx** [comment]
Hi @sudheerj ,
Yes, I confirm. Issue is only for v4.14.1. I tried with v4.13.0 and i work fine.
Hope you will be able to fix it.
Regards
M

**Comment by sudheerj** [comment]
The issue is fixed in 2.0.48 release. Please let us know for any issues.Thanks everybody for the details.
